packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -1337,8 +1392,19 @@ class RenderViewport extends RenderViewportBase<SliverPhysicalContainerParentDat
     final double reverseDirectionRemainingPaintExtent = centerOffset.clamp(0.0, mainAxisExtent);
     final double forwardDirectionRemainingPaintExtent = (mainAxisExtent - centerOffset).clamp(0.0, mainAxisExtent);
 
-    final double fullCacheExtent = mainAxisExtent + 2 * cacheExtent;
-    final double centerCacheOffset = centerOffset + cacheExtent;
+    double cacheMultiplier;
+    switch (cacheExtentStyle) {
+      case CacheExtentStyle.pixel:
+        _calculatedCacheExtent = cacheExtent;
+        cacheMultiplier = 2.0;
+        break;
+      case CacheExtentStyle.viewport:
+        _calculatedCacheExtent = mainAxisExtent;
```

Comment:
shouldn't this be
```dart
_calculatedCacheExtent = mainAxisExtent * cacheExtent;
cacheMultiplier = 2.0;
```

Reply:
That would also simplify the describeSemanticsClip as you can just use the _calculatedCacheExtent there?